core/orch_test.go

```diff
@@ -708,27 +700,88 @@ func TestProcessPayment_GivenReceiveTicketError_ReturnsError(t *testing.T) {
 	recipient.AssertNumberOfCalls(t, "RedeemWinningTicket", 2)
 }
 
-func TestProcessPayment_GivenInsufficientCredit_ReturnsErr(t *testing.T) {
+// Check that an Acceptable error increases the credit
+
+func TestProcessPayment_AcceptablePaymentError_IncreasesCreditBalance(t *testing.T) {
 	n, _ := NewLivepeerNode(nil, "", nil)
 	n.Balances = NewBalances(5 * time.Second)
 	recipient := new(pm.MockRecipient)
 	n.Recipient = recipient
 	orch := NewOrchestrator(n)
 	manifestID := ManifestID("some manifest")
-	sessionID := "some sessionID"
-	recipient.On("ReceiveTicket", mock.Anything, mock.Anything, mock.Anything).Return(sessionID, true, nil)
-	recipient.On("EV").Return(big.NewRat(1000000000, 1))
+	acceptableError := pm.NewMockReceiveError(errors.New("Acceptable ReceiveTicket error"), true)
+	recipient.On("ReceiveTicket", mock.Anything, mock.Anything, mock.Anything).Return("", false, acceptableError).Once()
+	assert := assert.New(t)
+	// Create a ticket where faceValue = EV instead of using 'defaultTicket'
+	// FaceValue is the expected balance increase
+	faceVal := big.NewInt(100)
+	ticketSenderParams := &net.TicketSenderParams{
+		SenderNonce: 456,
+		Sig:         pm.RandBytes(123),
+	}
 
-	recipient.On("RedeemWinningTicket", mock.Anything, mock.Anything, mock.Anything).Return(nil)
+	ticketParams := &net.TicketParams{
+		Recipient:         pm.RandBytes(123),
+		FaceValue:         faceVal.Bytes(),
+		WinProb:           new(big.Int).Sub(new(big.Int).Lsh(big.NewInt(1), 256), big.NewInt(1)).Bytes(),
+		RecipientRandHash: pm.RandBytes(123),
+		Seed:              pm.RandBytes(123),
+	}
+
+	sender := pm.RandBytes(123)
+	expirationParams := &net.TicketExpirationParams{
+		CreationRound:          5,
+		CreationRoundBlockHash: []byte{5},
+	}
+
+	payment := net.Payment{
+		TicketParams:       ticketParams,
+		Sender:             sender,
+		ExpirationParams:   expirationParams,
+		TicketSenderParams: []*net.TicketSenderParams{ticketSenderParams},
+	}
+	err := orch.ProcessPayment(payment, manifestID)
+	assert.Error(err)
+	assert.Zero(orch.node.Balances.Balance(manifestID).Cmp(new(big.Rat).SetInt(faceVal)))
+}
+
+// Check that an unacceptable error does NOT increase the credit
+func TestProcessPayment_UnacceptablePaymentError_DoesNotIncreaseCreditBalance(t *testing.T) {
+	n, _ := NewLivepeerNode(nil, "", nil)
+	n.Balances = NewBalances(5 * time.Second)
+	recipient := new(pm.MockRecipient)
+	n.Recipient = recipient
+	orch := NewOrchestrator(n)
+	manifestID := ManifestID("some manifest")
+	UnacceptableError := pm.NewMockReceiveError(errors.New("Unacceptable ReceiveTicket error"), false)
```

Comment:
"UnacceptableError" -> "unacceptableError"

Reply:
Fixed in https://github.com/livepeer/go-livepeer/pull/956/commits/f7b0485b60d135b363e3b5d78b6b0b7dadf1e4e9